syntax = "proto2";

package SimpleChat;

enum NetworkMessageType {
	USER_JOIN_REQUEST = 0;
	USER_JOIN_RESPONSE = 1;
	USER_LIST_REQUEST = 2;
	USER_LIST_RESPONSE = 3;
	USER_CHANGE = 4;
	
	CHAT_MESSAGE = 5;
	AUTHORIZE = 6;
	COMMAND = 7;
}

message NetworkMessageHeader {
	required string src = 1;
	required string dest = 2;
	required NetworkMessageType type = 3;
}

message NetworkMessage {
	required NetworkMessageHeader header = 1;
	required string serialized_data = 2;
}
